agent/config/parse_windows.go

```diff
@@ -0,0 +1,55 @@
+// +build windows
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+// parseGMSACapability is used to determine if gMSA support can be enabled
+func parseGMSACapability() bool {
+	envStatus := utils.ParseBool(os.Getenv("ECS_GMSA_SUPPORTED"), true)
+	if envStatus {
+		// If gMSA feature is explicitly enabled, check if container instance is domain joined.
```

Comment:
Still customer can override the environment variable and change the default behavior to be not supported. In those cases , we don't have the check the domain join.

Reply:
> isn't this check being done irrespective of the flag being explicitly enabled or not(true by default)?

Still customer can override the environment variable and change the default behavior to be not supported. In those cases , we don't have the check the domain join.

Reply:
Is there any use case that customers want to specifically disable it? Like if customer enables this feature, it will cause problems for their application?

From my understanding, even customers are not using this feature, we can still set this flag as true and advertise the capability. It will not affect anything.

Reply:
The capability should not be advertised when the environment variable is set but the container instance is not part of a valid active-directory.

Reply:
so Sid mentioned that the domainJoin is not enough to check whether it is joining the right domain. Like it can join the domain/AD which is not the right one for credentialSpec, if this is the case, is this check still adding value?

Reply:
@yumex93 I did not mention this. As discussed offline, the plan is to still have this check.